prog/dftb+/lib_dftbplus/main.F90

```diff
@@ -188,7 +188,7 @@ subroutine runDftbPlus(env)
       end if
     #:endif
       tWriteCharges = tWriteRestart .and. tMulliken .and. tSccCalc .and. .not. tDerivs&
-          & .and. maxSccIter > 1
+          & .and. maxSccIter > 1 .and. .not.allocated(reks)
```

Comment:
Wouldn't it be easier to check the allocation status of `qInput` instead?

Reply:
I agree, let's look for `allocated(qInput)` instead, and maybe as the first one in the condition list.